### PR TITLE
Jkralik/fix/projections

### PR DESCRIPTION
**Pull request by jkralik**

### CODE REVIEW

File: resource-directory/service/deviceMetadataProjection.go [hunk 1]
```diff
@@ -130,7 +131,7 @@ func (p *deviceMetadataProjection) Handle(ctx context.Context, iter eventstore.I
 				return err
 			}
 			if err := p.data.HandleDeviceMetadataUpdatePending(ctx, &e); err != nil {
-				return err
+				goto nextEvent
```
Comment (Danielius1922) [inline resource-directory/service/deviceMetadataProjection.go]: continue doesn't work for a for loop when called from an inner switch?

File: resource-aggregate/events/deviceMetadataSnapshotTaken.go [hunk 2]
```diff
@@ -131,25 +131,19 @@ func (e *DeviceMetadataSnapshotTaken) Handle(ctx context.Context, iter eventstor
 			if err := eu.Unmarshal(&s); err != nil {
 				return status.Errorf(codes.Internal, "%v", err)
 			}
-			if err := e.HandleDeviceMetadataSnapshotTaken(ctx, &s); err != nil {
-				return err
-			}
+			_ = e.HandleDeviceMetadataSnapshotTaken(ctx, &s)
```
Comment (Danielius1922) [inline resource-aggregate/events/deviceMetadataSnapshotTaken.go]: Feels weird to just ignore all these handle errors, wouldn't at least logging then be useful?

**Review by Danielius1922 — APPROVED** [review]

File: resource-directory/service/deviceMetadataProjection.go (hunk 1 above)
Comment (jkralik) [reply resource-directory/service/deviceMetadataProjection.go]: fixed

File: resource-aggregate/events/deviceMetadataSnapshotTaken.go (hunk 2 above)
Comment (jkralik) [reply resource-aggregate/events/deviceMetadataSnapshotTaken.go]: Removed error returns for snapshot handlers.

**Review by ondrejtomcik — APPROVED** [review]

**Review by Danielius1922 — APPROVED** [review]

**Pull request merged**